Automattic/wp-calypso
# State: Type notice actions and their options

#### Changes proposed in this Pull Request

As part of our work in #48408, we've been reduxifying a bunch of notices in the checkout area. Much of it being typed already, it came to our attention (thanks @sirbrillig for bringing it up) that those can cause errors because they're not typed. 

This PR introduces basic types for our notice action creators and their options, in order to make using them in TS code easier, and to help with decrypting the possible options that can be passed.

#### Testing instructions
* Verify build still works well.
* Smoke test notices throughout Calypso.
* Verify all tests still pass.
